### PR TITLE
unit test / code coverage improvements

### PR DESCRIPTION
**Pull request by HunterGerlach**
This PR has two contributions:

1. Add test for when stop token is found in middle of text

2. Add code coverage tooling and instructions
- Add pytest-cov via poetry
- Add necessary config files
- Add new make instruction for `coverage`
- Update README with coverage guidance
- Update minor README formatting/spelling

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -22,6 +22,7 @@ spacy = {version = "^3", optional = true}
 nltk = {version = "^3", optional = true}
 transformers = {version = "^4", optional = true}
 beautifulsoup4 = {version = "^4", optional = true}
+pytest-cov = "^4.0.0"
```
Comment (hwchase17) [inline pyproject.toml:25]: can we add this to the `tool.poetry.group.test.dependencies` group instead? shouldn't be needed for the built package right?

**Review by hwchase17 — COMMENTED** [review]
thank you for this!! 

sadly theres some merge conflicts with the README file now (I had split out the developer docs into a separate file, and removed some of the explantion since its a duplicate of what's in the docs). I'm happy to help resolve that if you want just lmk

**Comment by HunterGerlach** [comment]
@hwchase17 Merge conflicts resolved and poetry update made, Still looks like the automated build is failing for some reason/

**Comment by HunterGerlach** [comment]
I'm not a poetry expert, but I suspect it has something to do with my version of poetry building lock files in v2 format, while the original is still in v1.

**Comment by HunterGerlach** [comment]
Yup - that was it :tada:

**Comment by hwchase17** [comment]
> Yup - that was it 🎉
> 
> Created a separate PR to specifically address the version bump...
> 
> #328

got it - thanks for debugging! will just merge this one. locally i was on poetry version 1.2.2 (which has forward compatability with v2 format) so this wont actually affect me (and other people on that version)

Thanks for doing this!

**Review by hwchase17 — APPROVED** [review]

**Pull request merged**